src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1014,15 +1015,16 @@ def _get_enable_windows_gmsa(self, enable_validation: bool = False, **kwargs) ->
         """
         # read the original value passed by the command
         enable_windows_gmsa = self.raw_param.get("enable_windows_gmsa")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.windows_profile and
-            hasattr(self.mc.windows_profile, "gmsa_profile") and  # backward compatibility
-            self.mc.windows_profile.gmsa_profile and
-            self.mc.windows_profile.gmsa_profile.enabled is not None
-        ):
-            enable_windows_gmsa = self.mc.windows_profile.gmsa_profile.enabled
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
```

Comment:
FYI: WindowsGmsa tests can be run with setting `AKSWindowsGmsaPreview` in AKSHTTPCustomFeatures.

Reply:
Yeah, we are using the custom header to test this feature in our live test case since [PR #4161](https://github.com/Azure/azure-cli-extensions/pull/4161).